src/stories/containers/cu-table/cu-table-2.mvvm.ts

```diff
@@ -211,13 +213,41 @@ export const useCoreUnitsTableMvvm = () => {
     return sortDataFunction;
   }, [headersSort, sortColumn]);
 
+  const giveWeightByStatus = (status: CuStatusEnum) =>
+    status === CuStatusEnum.Accepted
+      ? 5
+      : status === CuStatusEnum.FormalSubmission
+      ? 4
+      : status === CuStatusEnum.RFC
+      ? 3
+      : status === CuStatusEnum.Obsolete
+      ? 2
+      : status === CuStatusEnum.Rejected
+      ? 1
+      : 0;
+  const dataWithStatus = filteredData?.map((item) => ({
+    ...item,
+    status: giveWeightByStatus(getStautsMip39AccetedOrObsolete(item)),
+  }));
```

Comment:
I'm not 100% sure, but `dataWithStatus` will be a new array on every render and it is used as a dependency in the `useMemo` of the next sentence so on every render the `useMemo` will be revalidating the value and doing the calculation. You should memorize this value too

Reply:
ok, you are right